backend/app/api/endpoints/devices.py

```
@@ -362,3 +380,93 @@ async def trigger_device_upgrade(
             status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
             detail=f"Failed to trigger upgrade: {str(e)}",
         )
+
+
+@router.post("/{device_id}/commands", response_model=DeviceCommandResponse)
+async def execute_device_command(
+    device_id: str,
+    request: DeviceCommandRequest,
+    db: Session = Depends(get_db),
```

Ruff (0.15.13)

[warning] Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)